packages/database/src/core/util/libs/parser.ts

```diff
@@ -131,6 +131,8 @@ export const parseURL = function(
       subdomain = parts[0].toLowerCase();
     } else if (parts.length === 2) {
       domain = parts[0];
+    } else if (parts[0].split(':')[0].toLowerCase() === 'localhost') {
+      subdomain = 'localhost';
```

Comment:
Can you move the port extraction logic from line 136 up so that it happens before this if-statement? Then we won't have to double parse the port number.

Then I would also suggest that instead of assigning 'localhost' to `subdomain` you should assign it to `domain`. A subdomain would be something like 'foo' in 'foo.localhost'.